src/ServiceControl.Monitoring/Infrastructure/EndpointTracker.cs

```diff
@@ -1,15 +1,17 @@
 ﻿namespace ServiceControl.Monitoring.Infrastructure
 {
+    using System;
     using System.Threading.Tasks;
     using Messaging;
     using NServiceBus;
     using NServiceBus.Metrics;
 
     public class EndpointTracker : IHandleMessages<MetricReport>, IHandleMessages<LongValueOccurrences>, IHandleMessages<Occurrences>
     {
-        public EndpointTracker(EndpointRegistry endpointRegistry)
+        public EndpointTracker(EndpointRegistry endpointRegistry, EndpointInstanceActivityTracker activityTracker)
```

Comment:
I thought that there was one handler for all the messages but then I saw this that handles them explicitly. My assumption was to do not change everything and just attach to the thing that registers endpoints' instances in a registry. This might be a subject to change in the future (but then again, it will cover the registry and the activity tracker at the same time).

Does it answer you question?